draft-ietf-httpbis-no-vary-search.md

```diff
@@ -390,6 +392,15 @@ So, for example, given any non-default value for `No-Vary-Search`, such as `No-V
   <dd><tt>+</tt> and <tt>%20</tt> are both parsed as U+0020 SPACE</dd>
 </dl>
 
+# Caching
+
+{:aside}
+> A more thorough discussion of how the above algorithms are used in HTTP caches {{HTTP-CACHING}} generally is likely merited.
+
+Servers should send no more than one distinct non-empty value for the `No-Vary-Search` field in response to requests for a given pathname. Cache implementations may fail to reuse a stored response whose target URI is not equal to the presented target URI ({{Section 7.1 of HTTP}}) even though the two are equivalent modulo URL search variance for the variance obtained ({{obtain-a-url-search-variance}}) from the stored response, if the cache has more recently stored a response with a target URI which is equal, excluding the query, and a different non-empty value for the `No-Vary-Search` field.
+
+This permits cache implementations to reduce the number of accesses required to search for a response (to a single search for an exact match, plus one more if the cache has seen `No-Vary-Search` used for that path, using a simplified query string). Since such implementations might "miss" some stored responses that could otherwise have been reused, it is useful for servers to avoid sending different values where possible.
```

Comment:
It's still not obvious to me how this works. Maybe you could spell it out with pseudo-code? And annotate the steps which depend on the optional step above? I tried to write some myself but got confused.

Reply:
I previously described it here: https://github.com/WICG/nav-speculation/issues/210#issuecomment-2364516756

Is that helpful enough? I'm trying not to mandate anything in particular, just make algorithms along those lines permissible.